### PR TITLE
Xcode 10 + using xcconfig for project settings

### PR DESCRIPTION
**Pull request by bpoplauschi**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

### Pull Request Description

Xcode 10 - upgraded schemes, using xcode10 Travis image, removed some deprecated methods from our demos + formatting

Using xcconfig files - easier to maintain all the settings appart from target files, less conflicts, can reuse the values

### CODE REVIEW

**Comment by bpoplauschi** [comment]
The build fails because the `pod lib lint` fails on Xcode 10 - not related to this PR, but I will fix it.

**Comment by dreampiggy** [comment]
@bpoplauschi Sounds good. But one more question, does this changes be compatible for Xcode 8 (which means, after this change is merged, can Xcode 8 user still build it without any issue ?). Current 4.x seems be a compatible version supports both Xcode 8 && iOS 7...

If there are something break effect, maybe we can hurry up to release 5.x. It takes too much of times 😅  . Current pending PR for 5.x, which does not introduce API breaking should be considered in low priority. Because we can fix it in 5.0.1 or future after we release the official 5.0.0.

**Comment by bpoplauschi** [comment]
@dreampiggy 
- this PR keeps compatibility with the OS versions, but because of one setting (`SWIFT_VERSION=4.2`) it requires Xcode 10. I agree we can keep this 4.x compatibility with older versions and move this PR to 5.x
- I also want to push and release 5.0. I agree. Let's look again at the list of remaining things for 5.0 and just get it out

**Comment by bpoplauschi** [comment]
Agree @dreampiggy it is taking too long. I will change this PR for 5.x branch and create another one for master so we can support Xcode 10 and iOS 12 in 4.4.3 - see #2495

**Comment by bpoplauschi** [comment]
I will: rebase on top of 5.x, then change the branch target here

**Comment by bpoplauschi** [comment]
Build passes. You can review again guys. This time if against the 5.x branch.

File: Tests/Tests/SDWebImageDownloaderTests.m [hunk 1]
```diff
@@ -181,9 +181,9 @@ - (void)test11ThatCancelWorks {
 
 - (void)test12ThatWeCanUseAnotherSessionForEachDownloadOperation {
     XCTestExpectation *expectation = [self expectationWithDescription:@"Owned session"];
-    NSURL *imageURL = [NSURL URLWithString:kTestJPEGURL];
```
Comment (dreampiggy) [inline Tests/Tests/SDWebImageDownloaderTests.m]: Small question, does this variable naming cause a Xcode warning ? 😕 Anyway it's OK, just a little curious

Comment (bpoplauschi) [reply Tests/Tests/SDWebImageDownloaderTests.m]: It's not that obvious, but the name was `imageURL` and that was shadowed by a param with the same name in the `addHandlersForProgress` block below it:
```obj-c
SDWebImageDownloaderOperation *operation = [[SDWebImageDownloaderOperation alloc] initWithRequest:request
                                                                                            inSession:nil
                                                                                              options:0];
[operation addHandlersForProgress:^(NSInteger receivedSize, NSInteger expectedSize, NSURL *imageURL) {
} completed:^(UIImage * _Nullable image, NSData * _Nullable data, NSError * _Nullable error, BOOL finished) {
    if (image && data && !error && finished) {
        [expectation fulfill];
    } else {
        XCTFail(@"Something went wrong");
    }
}];
```
All the other changes are similar, there are blocks where the param is shadowed

Comment (dreampiggy) [reply Tests/Tests/SDWebImageDownloaderTests.m]: OK.

**Comment by dreampiggy** [comment]
LGTM...I run the target branch with Xcode 10.1. Xcode config can avoid massive configuration across different target, it's a tool for the large project which contains multiple targets but share some common configs.

**Comment by bpoplauschi** [comment]
@dreampiggy do you have second thoughts about merging this? :)

**Comment by dreampiggy** [comment]
No. I don't check each of configuration in xcconfig, but if you can ensure all their value are from current Xcode's project config, it's OK to swtich to the xconfig to manage them. It's text-based and more easy for git to show the diff (instead of that `pbproj` files which is hard to read)

**Comment by bpoplauschi** [comment]
Then please merge it, I don't want to merge my own PR :)

**Pull request merged**